### DISCUSSION TITLE
Implovement idea - report wrong file-digest as error, instead of "AppxManifest.xml does not match'

### DISCUSSION THREAD

**Discussion by MindScriptAct** [vcsjones/AzureSignTool#278] [Ideas]
Hello, 

 I lost 4 days of my life trying to solve "The Publisher Identity in the AppxManifest.xml does not match the subject on the certificate."
 this problem : 

 I tried escaping characters, making identity shorter... longer... everything...
 
The problem was - wrong --file-digest setting.  (I used SHA384 instead of SHA256)

It would help a ton if better error is produced.

Thank you for your time.

full log : 

trce: AzureSignTool.SignCommand[0]
      => File: C:\aWorkSpace\integrostim\IntegroStimApp\Publish\IntegroStimPlayer.WinUI_0.0.1.0_x64_Debug_Test\IntegroStimPlayer.WinUI_0.0.1.0_x64_Debug.msix
      Calling SignerSignEx3 with flags: SPC_EXC_PE_PAGE_HASHES_FLAG, SIGN_CALLBACK_UNDOCUMENTED
fail: AzureSignTool.SignCommand[0]
      => File: C:\aWorkSpace\integrostim\IntegroStimApp\Publish\IntegroStimPlayer.WinUI_0.0.1.0_x64_Debug_Test\IntegroStimPlayer.WinUI_0.0.1.0_x64_Debug.msix
      The Publisher Identity in the AppxManifest.xml does not match the subject on the certificate.
fail: AzureSignTool.SignCommand[0]
      => File: C:\aWorkSpace\integrostim\IntegroStimApp\Publish\IntegroStimPlayer.WinUI_0.0.1.0_x64_Debug_Test\IntegroStimPlayer.WinUI_0.0.1.0_x64_Debug.msix
      Signing failed with error 8007000B.

**Comment by TiberiusDRAIG** [comment]
Just as a note on this for people getting here from Google and the like: this error is also reported if you try to sign an MSIX on Windows 10 or a Windows 10 based OS (e.g. Server 2022) when the package uses [MSIX Persistent Identity](https://learn.microsoft.com/en-us/windows/msix/package/persistent-identity) (i.e. you're transitioning between certificates where the `Subject` has changed). In this scenario you must use a Windows 11 (or likely later) based OS to perform signing.

See: https://github.com/vcsjones/AzureSignTool/issues/304.
